statsmodels/genmod/generalized_linear_model.py

```diff
@@ -1150,8 +1145,6 @@ def fit_constrained(self, constraints, start_params=None, **fit_kwds):
         res._results.constraints = lc
         res._results.k_constr = k_constr
         res._results.results_constrained = res_constr
-        # TODO: the next is not the best. history should bin in results
-        res._results.model.history = res_constr.model.history
```

Comment:
Is `res_constr.model.history` always empty?

Reply:
> that means [...] should work as a replacement

I'm not entirely clear, is this actionable for the OP?

The only strong opinion I have is that the Model instance should not be changed by the call to `fit`.